src/app/public/testing/sort/sort-fixture.ts

```diff
@@ -0,0 +1,151 @@
+
+import {
+  SkySortFixtureMenu
+} from './sort-fixture-menu';
+
+import {
+  SkySortFixtureMenuItem
+} from './sort-fixture-menu-item';
+
+/**
+ * Provides information for and interaction with a SKY UX sort component.
+ * By using the fixture API, a test insulates itself against updates to the internals
+ * of a component, such as changing its DOM structure.
+ */
+export class SkySortFixture {
+
+  /**
+   * The active sort menu item, if one exists. Menu items are only available when the menu dropdown
+   * is open. If the menu dropdown is closed, this property will be undefined.
+   */
+  public get activeMenuItem(): SkySortFixtureMenuItem {
+    return this.menuItems?.find(x => x.isActive);
+  }
+
+  /**
+   * The sort menu's properties.
+   */
+  public get menu(): SkySortFixtureMenu {
+    return {
+      buttonText: SkyAppTestUtility.getText(this.getSortButtonTextSpan()),
+      isOpen: this.getDropdownMenuEl() !== null
+    };
+  }
+
+  /**
+   * The properties of each sort menu item. Menu items are only available when the menu dropdown
+   * is open. If the menu dropdown is closed, this property will be undefined.
+   */
+  public get menuItems(): SkySortFixtureMenuItem[] {
+    // Return undefined when we can't determine what the options are.
+    // We do this to avoid any confusion with an empty set of options.
+    if (!this.menu.isOpen) {
+      return undefined;
+    }
+
+    return this.getSortItems()
+      .map((item: HTMLElement) => {
+        const itemButton = item.querySelector('button');
+
+        return {
+          isActive: item.classList.contains('sky-sort-item-selected'),
+          text: SkyAppTestUtility.getText(itemButton)
+        };
+      });
+  }
+
+  private _debugEl: DebugElement;
+
+  constructor(
+    private fixture: ComponentFixture<any>,
+    skyTestId: string
+  ) {
+    this._debugEl = SkyAppTestUtility.getDebugElementByTestId(fixture, skyTestId, 'sky-sort');
+  }
+
+  /**
+   * Ensures the sort menu is open and selects the menu item with the specified text,
+   * if a matching item is available.
+   * @param menuItemText The text of the menu item to select.
+   */
+  public async selectMenuItem(menuItemText: string): Promise<void> {
+
+    // make sure the sort menu is open
+    if (!this.menu.isOpen) {
+      await this.toggleMenu();
+    }
+
+    // find the requested menu item
+    const items = this.getSortItems();
+    const targetItem = items.find((item: HTMLElement) => {
+      return SkyAppTestUtility.getText(item) === menuItemText;
+    });
+
+    // if we found the item, select it
+    if (targetItem) {
+      // we've got the '.sky-sort-item' div, but we want to click it's child button element
+      const targetButton = targetItem.querySelector('button');
+      targetButton.click();
+
+      this.fixture.detectChanges();
+      return this.fixture.whenStable();
+    }
+  }
+
+  /**
+   * Toggles the sort dropdown menu open or closed.
+   */
+  public async toggleMenu(): Promise<void> {
```

Comment:
I tend to prefer more "functional" methods such as `openMenu` and `closeMenu` since the outcome is more obvious and deliberate. The user must know the current state of the menu before calling `toggleMenu` in order to predict the outcome, which I feel is detrimental to the experience.

Reply:
The action here is the same regardless. I'd have to create two separate methods and throw some error if the menu was already open or closed. I can certainly do that if it's critical, but this would not be the first place I've seen this done in the code base.

Reply:
Anyways, the change isn't a big deal so I'm doing it right now.